### DISCUSSION TITLE
[CT-2311] Integrate test results (run_results.json) in catalog

### DISCUSSION THREAD

**Discussion by matthieucan** [dbt-labs/dbt-docs#404] [Ideas]
### Describe the feature
Having test results, read from `run_results.json` integrated in the dbt docs would be amazing.

It is unclear what it should look like, but examples include:
- An additional section in the "Model" page listing the associated tests and their status (pass or fail).
- Additional metadata next to a given Column listing associated test results.

### Describe alternatives you've considered
Elementary Data (https://www.elementary-data.com/) can generate a beautiful static page out of test results. They have many interesting features, such as historization of results, but unfortunately a different web page is created, not integrated in dbt docs' `index.html`.

### Additional context
Not database specific, to the best of my knowledge all necessary metadata is already present in `run_results.json`, and should be retrieved and exposed.

### Who will this benefit?
All users who run tests on a schedule, and are happy to update the catalog after running tests. This adds data observability to augment all the metadata already exposed by dbt docs.

### Are you interested in contributing this feature?
Yes. Starting an early discussion now to get feedback from maintainers, to understand if it's desirable and/or on the roadmap.

**Comment by dbeatty10** [comment]
This is a really interesting Big Idea @matthieucan ! 💡

I'm going to promote this to a Discussion to gather opinions and feedback.
